edifact/models/message.py

```diff
@@ -1,198 +1,261 @@
+        return cls(qualifier_and_code, timestamp)
+
+    @property
+    def key(self):
+        return 'DTM'
+    @property
+    def value(self):
+        formatted_date_time = self.timestamp.strftime(self.date_time_format)
+        return f'{self.type_code}:{formatted_date_time}:{self.format_code}'
+    def pre_validate(self):
+        self._required('type_code')
+        self._required('format_code')
+        self._required('date_time_format')
+        self._required('timestamp')
+class Reference(Segment):
+    def __init__(self, qualifier: str, reference: (None, int, str)):
+        self.qualifier = qualifier
+        self.reference = reference
+    @property
+    def key(self):
+        return 'RFF'
+    @property
+    def value(self):
+        return f'{self.qualifier}:{self.reference}'
+    def pre_validate(self):
+        self._required('qualifier')
+        self._required('reference')
+
+
+class ReferenceTransactionType(Reference):
+
+    class TransactionType(enum.Enum):
+        ACCEPTANCE = 'G1'
+        AMENDMENT = 'G2'
+        REMOVAL = 'G3'
+        DEDUCTION = 'G5'
+    def __init__(self, transaction_type: TransactionType):
+        super().__init__(qualifier='950', reference=transaction_type.value)
+
+    @classmethod
+    def from_string(cls, message_line: str) -> Segment:
+        """
+        generates edifact segment from given edifact message line
+        """
+        rff_match = re.match(RFF_PATTERN, message_line)
+        transaction_type = rff_match.group('transaction_type')
+
+        transaction_types = {}
+        for t in ReferenceTransactionType.TransactionType:
+            transaction_types[t.value] = t
+        transaction_type_code = transaction_types.get(transaction_type)
+
+        return cls(transaction_type_code)
+
+
+class ReferenceTransactionNumber(Reference):
+
+    def __init__(self, reference: (int, None) = None):
+        super().__init__(qualifier='TN', reference=reference)
+
+    @classmethod
+    def from_string(cls, message_line: str) -> Segment:
+        """
+        generates edifact segment from given edifact message line
+        """
+        rff_tn_match = re.match(RFF_TN_PATTERN, message_line)
+        reference = int(rff_tn_match.group('transaction_number'))
+        return cls(reference)
+
+    def pre_validate(self):
+        self._required('qualifier')
+
+    def _validate_stateful(self):
+        self._required('reference')
+class SegmentGroup(Segment):
+    def __init__(self, segment_group_number: int):
+        self.segment_group_number = segment_group_number
+
+    @classmethod
+    def from_string(cls, message_line: str) -> Segment:
+        """
+        generates edifact segment from given edifact message line
+        """
+        sg_match = re.match(SG_PATTERN, message_line)
+
+        sg0 = sg_match.group('segment_group_number_0')
+        sg1 = sg_match.group('segment_group_number')
+
+        # TODO: sg0 is zero-padded, ignoring this check for now
```

Comment:
I think this TODO and commented code can be removed

Reply:
@mattd-kainos, I think this code should be here, but not commented. The question is do we want to check if the number in sg0 is the same as in the sg1? The zero-padding does not matter, I can take int() of both.